lib/client/util/__mocks__/Geolocation.js

```diff
@@ -0,0 +1,37 @@
+var path = require.requireActual('../Geolocation');
+
+var geoMock = jest.genMockFromModule('../Geolocation');
+
+var _coords = null;
+var _enabled = false;
+function __setLocation(lat, lon) {
+  _enabled = true;
+  _coords = { latitude: lat, longitude: lon };
+}
+
+function __enable() {
+  _enabled = true;
+}
+
+function __disable() {
+  _coords = null;
+  _enabled = false;
+}
+
+function getCurrentPosition(successCallback, errorCallback) {
+  if(_enabled) {
+    if(_coords) {
+      successCallback({ coords: _coords });
+    }
```

Comment:
so do i understand the article correctly that they're saying that just because your code integrates with an async module, you can still test your code as if it were not async? The main issue I have is that if something is integrating with async code, it should always behave in an async way even if it has, for example, cached data and can instantly return it. A consumer should be able to rely on code they use to be async if it takes a callback.

Anyway, I didn't know that there were extra things you have to do in Jest to get timers to run as they would normally. That's kind of weird. It makes me wonder what breadth of functionality Jest provides- in other words, what other magic is involved.

The main reason I suggested the setTimeout is because we can leverage the async nature to test the loading state. I think the mock should match the real api and functionality as closely as possible. I definitely don't think the mock should be synchronous.

I'm still willing to discuss it further though, and willing to work on this over time instead of blocking a PR on the discussion.